tokio-stream/src/stream_close.rs

```diff
@@ -0,0 +1,108 @@
+use crate::Stream;
+use pin_project_lite::pin_project;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+pin_project! {
+    /// Stream that returns None when the stream is exhausted. To be used with StreamMap to know when a stream ends.
+    /// # Example
+    /// Using `StreamNotifyClose` to handle closed streams with `StreamMap`
+    /// ```no_run
+    /// use tokio_stream::{Stream, StreamExt, StreamMap, StreamNotifyClose};
+
+    /// struct Test {
+    ///     state: i32,
+    /// }
+
+    /// impl Stream for Test {
+    ///     type Item = i32;
+
+    ///     fn poll_next(mut self: Pin<&mut Self>, _cx: &mut Context<'_>) -> Poll<Option<i32>> {
+    ///         let val = self.state;
+    ///         let res = match val {
+    ///             0 => Poll::Ready(Some(0)),
+    ///             1 => Poll::Ready(Some(1)),
+    ///             2 => Poll::Ready(None),
+    ///             _ => unreachable!("shouldn't get here"),
+    ///         };
+
+    ///         self.state += 1;
+    ///         res
+    ///     }
+    /// }
```

Comment:
This is fine for the test, but too complicated to be an example. You should use a simpler `Stream` instead. For example, you could use `tokio_stream::iter(vec![0, 1])`.